### PR TITLE
Pull Request for Issue1698a: Developing a view for sample chamber controls

### PR DESCRIPTION
**Pull request by helfrij**
The original specification for #1698 was to have a completely automated vacuum/venting system for SGM, but for a couple of reasons it isn't possible to complete and test before the deadline. Instead, developed a simple view that displays information for the sample chamber and related controls, and allows a user to manipulate them manually.

### CODE REVIEW

File: source/beamline/SGM/SGMSampleChamberVacuum.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#define SGMSAMPLECHAMBERVACUUM_H
+
+#include "beamline/AMSingleEnumeratedControl.h"
+
+#define SGMSAMPLECHAMBERVACUUM_VENTED_MIN 760.0
+#define SGMSAMPLECHAMBERVACUUM_VENTED_MAX 1000.0 // Has to be larger than any value we would encounter.
+#define SGMSAMPLECHAMBERVACUUM_ROUGHVACUUM_MIN 0.001
+#define SGMSAMPLECHAMBERVACUUM_ROUGHVACUUM_MAX 10.0
+#define SGMSAMPLECHAMBERVACUUM_HIGHVACUUM_MIN 0.000000001 // Has to be smaller than any value we would encounter.
+#define SGMSAMPLECHAMBERVACUUM_HIGHVACUUM_MAX 0.001
```
Comment (davidChevrier) [inline source/beamline/SGM/SGMSampleChamberVacuum.h]: I'm guessing SGMSAMPLECHAMBERVACUUM_HIGHVACUUM_MAX and SGMSAMPLECHAMBERVACUUM_ROUGHVACUUM_MIN are equal?

File: sgmCommon.pri [hunk 2]
```diff
@@ -27,7 +27,16 @@ HEADERS += \
     source/beamline/SGM/SGMBypassLadder.h \
     source/beamline/SGM/SGMXASLadder.h \
 	source/ui/SGM/SGMLaddersView.h \
-    source/beamline/SGM/energy/SGMExitSlitSupport.h
+    source/beamline/SGM/energy/SGMExitSlitSupport.h \
+    source/beamline/SGM/SGMVATValve.h \
+    source/beamline/SGM/SGMSampleChamberVacuum.h \
+    source/ui/SGM/SGMVATValveView.h \
+    source/beamline/SGM/SGMTurboPump.h \
+    $$PWD/source/beamline/SGM/SGMSampleChamber.h \
```
Comment (davidChevrier) [inline sgmCommon.pri]: I don't think the $$PWD will hurt anything ... but I wonder why it was necessary.

Comment (helfrij) [inline sgmCommon.pri]: Not sure! I upgraded my Qt Creator halfway through this project and I think that's when $$PWD started being added to class paths.

File: source/beamline/SGM/SGMSampleChamberVacuum.h (hunk 1 above)
Comment (helfrij) [inline source/beamline/SGM/SGMSampleChamberVacuum.h]: Hmmm. Do you think those ranges make sense? As written:

High vacuum --> very small - 0.001
Rough vacuum --> 0.001 - 10
Vented --> 760 - very large

High vacuum and rough vacuum range limits are adjacent, but the vented and rough vacuum limits are not. I think this means that the control will have the value "Unknown" when between these two states.

Comment (davidChevrier) [inline source/beamline/SGM/SGMSampleChamberVacuum.h]: My sense is that the numbers are reasonable. I would say that "high vacuum" and "rough vacuum" are adjacent ranges ... but "vented" is fairly far from "rough vaccum". I'm not really sure what's supposed to be in the middle but it's not a state you want to stay in, that's for sure.

Comment (helfrij) [inline source/beamline/SGM/SGMSampleChamberVacuum.h]: Okay, cool. Just checking. Thanks @davidChevrier!

**Comment by davidChevrier** [comment]
Time Estimate: 15 minutes

**Pull request merged**